### PR TITLE
COR-303: Add ISO locale in @clerk/localizations + Intl.ListFormat 

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

### CODE REVIEW

File: packages/clerk-js/src/ui/hooks/usePasswordComplexity.ts [hunk 1]
```diff
@@ -68,7 +69,7 @@ export const usePasswordComplexity = (config: UsePasswordComplexityConfig, callb
 
   const [password, _setPassword] = useState('');
   const [failedValidations, setFailedValidations] = useState<ComplexityErrorMessages>({});
-  const { t } = useLocalizations();
+  const { t, locale } = useLocalizations();
```
Comment (anagstef) [inline packages/clerk-js/src/ui/hooks/usePasswordComplexity.ts:72]: 🔧  The `useLocalizations` hook could return a boolean `canUseListFormat` value that uses the utility internally (?) Just a suggestion.

**Review by anagstef — APPROVED** [review]

File: packages/clerk-js/src/ui/localization/makeLocalizable.tsx [hunk 2]
```diff
@@ -86,7 +89,7 @@ export const useLocalizations = () => {
     );
   };
 
-  return { t, translateError };
+  return { t, translateError, locale: localization?.locale || defaultResource?.locale };
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/localization/makeLocalizable.tsx:92]: 💯

File: packages/types/src/localization.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ export type LocalizationValue = string;
 export type LocalizationResource = DeepPartial<_LocalizationResource>;
 
 type _LocalizationResource = {
+  locale: 'en-US' | 'fr-FR' | 'de-DE' | 'da-DK' | 'es-ES' | 'it-IT' | 'nl-NL' | 'nl-BE' | 'pt-BR' | 'ru-RU' | 'sv-SE';
```
Comment (desiprisg) [inline packages/types/src/localization.ts:17]: Should we allow all values here for any custom implementations without a localization preset from us?

Comment (panteliselef) [reply packages/types/src/localization.ts:17]: @anagstef @anagstef would you like to just append string to the type so  the auto complete stays there but we continue to support any other locale

Comment (anagstef) [reply packages/types/src/localization.ts:17]: @panteliselef Sounds good for me!

Comment (desiprisg) [reply packages/types/src/localization.ts:17]: Sure, if that works. We just don't want any ts errors popping up if they use some other locale.

Comment (panteliselef) [reply packages/types/src/localization.ts:17]: Valid request, I will update the PR before merging

**Review by desiprisg — APPROVED** [review]

**Pull request merged**